src/amo/components/Category/index.js

```diff
@@ -67,11 +68,17 @@ export function mapStateToProps(state, ownProps) {
     { ...state.search.filters, page: parsePage(state.search.page) },
     { ...filters, page: queryParams.page },
   );
+
+  const hasFetched = state.categories.hasFetched;
```

Comment:
I don't think you need to introduce this new state attribute since you already have it as `state.categories.loading`. My hunch is that you added it because the new `loading` prop was not taking effect (I explained that below).